### ISSUE TITLE
module docs to not appear to exist on github

### ISSUE THREAD

**Issue by hobson** [ansible/ansible#8013]
##### Issue Type:

Bug Report
Documentation Report
##### Ansible Version:

ansible 1.6.3
##### Environment:

Fedora 16-19
##### Summary:

Bug Report: 
There's no " Edit on GitHub" link on any of the module pages on docs.ansible.com, nor can I find that erroneous phrase in github.com/ansible on the `release` and `devel` branches I've searched.

Documentation Report:
The [copy module docs](http://docs.ansible.com/copy_module.html) erroneously state that `state=file` means "the file will NOT be created...see the copy module..."
##### Steps To Reproduce:

https://github.com/ansible/ansible/tree/devel/docsite/rst/modules is empty

"Search this repo" on github/ansible for "file will NOT be created" comes up empty
##### Expected Results:

`.rst` files for each of the modules in `docsite/rst/modules`?
`DOCUMENTATION = '''` within `libraries/files/copy.py` containing the phrase "NOT be created"
##### Actual Results:

Can't find orPR the docs for any core module.

**Comment by jimi-c** [comment]
The docs are in the git repository, contained within the docstring for each module. If you have any further questions regarding this, please direct them to the mailing list. Thanks!

**Comment by hobson** [comment]
@jimi-c The [docs in the git repository](https://github.com/ansible/ansible/blob/devel/library/files/copy) do not match the [docs online](http://docs.ansible.com/copy_module.html) and both have errors.

Also, the "Edit on Github" button is missing from the docs online.

**Comment by jimi-c** [comment]
@hobson we try and do regular pushes to update those, but apparently we're behind. I can update them now.

**Comment by hobson** [comment]
@jimi-c https://github.com/jimi-c They are still mismatched and there's
still no way for me to help correct the docs for modules (like there is for
the rest of the ansible docs with that nifty "edit on github" link).

**Comment by jimi-c** [comment]
At this time, we do not have any plans to add an "Edit on Github" button for module docs. Which doc is mismatched?

**Comment by hobson** [comment]
@jimi-c The [file.copy docstring](https://github.com/ansible/ansible/blob/devel/library/files/copy) does not match the [html docs](http://docs.ansible.com/copy_module.html). Notice there are only 7 options in the docstring and about two dozen listed in the html table titled "Options". And the phrase "the file will NOT be created" exists in the HTML docs but does not exist anywhere in the ansible source code repository (that I can find).

**Comment by jimi-c** [comment]
The extra options come from this: `extends_documentation_fragment: files`. There are common options across many of the file-like modules, so rather than duplicating that we have documentation fragments (they also exist for things like the AWS modules, etc.). Those are also stored in the github repo, in `lib/ansible/utils/module_docs_fragments/`.

**Comment by hobson** [comment]
Thanks @jimi-c. The `state` option has unique behavior for each command in the file module and the docs for it should not be shared between them. For example, the shared fragment is incorrect here: "If [state=]file, the file will NOT be created if it does not exist, see the copy or template module if you want that behavior." This is not the behavior of this option for the `file.copy` module that I have observed and the sentence actually self-refers to the docs for the `copy` module (which is where this text was found).  So this sentence doesn't belong in a shared doc fragment for all the `file.*` modules.

Is it enough to "override" a docstring for a particular option (`state`) in a particular module (`file.copy`)? Or do I need to delete this offending option row from the shared fragment, and then copy the removed text into all the other docstrings that need the old text, and then edit the removed text before inserting into in the docstring for the `file.copy` module?

TLDR: Philosophical comment. 
That architecture is DRY and clever, but it seems to reduce one of the main benefits of putting docs within source code--allowing newbies reading the code to understand it. An ansible-newbie human like me can't read the ansible modules and learn about ansible behavior--I have to first figure out ansible's file layout and docstring rendering logic.

**Comment by jimi-c** [comment]
Yes, if it's something that can't be generalized, it shouldn't be in that common code.

**Comment by hobson** [comment]
@jimi-c, To answer my own question, it looks like the documentation renderer uses a `dict.update`. So nullifing the content for the `state` attribute within the `file.copy` docstring should fix this bug.  Alternatively, it seems you're suggesting `extends_documentation_fragment` should not be used for the copy command.
